inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/filebased/ConfigurationDirectoriesPoller.java

```diff
@@ -0,0 +1,148 @@
+
+    private ScheduledFuture<?> pollingTask = null;
+
+    private List<DirectoryPropertySourceChangeChecker> filePollers;
+
+    public ConfigurationDirectoriesPoller() {
+        super("config.fileBased");
+    }
+
+    @Override
+    protected boolean checkEnabledForConfig(InspectitConfig conf) {
+        FileBasedConfigSettings fileBased = conf.getConfig().getFileBased();
+        return fileBased.isWatch() && fileBased.getFrequency().toMillis() > 0;
+    }
+
+    @Override
+    protected boolean doEnable(InspectitConfig conf) {
+        log.info("Starting config directory polling service..");
+        env.readPropertySources(propertySources ->
+                filePollers = propertySources.stream()
+                        .filter(ps -> ps instanceof DirectoryPropertySource)
+                        .map(ps -> (DirectoryPropertySource) ps)
+                        .map(ps -> createCheckerForDirectory(ps))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toList())
+        );
+        if (filePollers.isEmpty()) {
+            log.info("No directories to watch registered, terminating poll service.");
+            return false;
+        } else {
+            long freqMS = conf.getConfig().getFileBased().getFrequency().toMillis();
+            pollingTask = executor.scheduleWithFixedDelay(this::pollDirectories, freqMS, freqMS, TimeUnit.MILLISECONDS);
+            return true;
+        }
+    }
+
+    private void pollDirectories() {
+        for (val poller : filePollers) {
+            poller.checkForChangesAndReloadIfRequired();
+        }
+    }
+
+
+    @Override
+    protected boolean doDisable() {
+        if (pollingTask != null) {
+            log.info("Stopping config directory polling service.");
+            pollingTask.cancel(true);
+            for (val poller : filePollers) {
+                try {
+                    poller.destroy();
+                } catch (Exception e) {
+                    log.error("Error destroying poller!", e);
+                }
+            }
+        }
+        return true;
+    }
+
+    private Optional<DirectoryPropertySourceChangeChecker> createCheckerForDirectory(DirectoryPropertySource dps) {
+        val result = new DirectoryPropertySourceChangeChecker();
+        try {
+            result.init(dps);
+            return Optional.of(result);
+        } catch (Exception e) {
+            log.error("Error creating watcher for directory!", e);
+            return Optional.empty();
+        }
+    }
+
+    private class DirectoryPropertySourceChangeChecker {
+
+        private FileAlterationObserver observer;
+        private DirectoryPropertySource dirToWatch;
+        private boolean anyFileChanged = false;
+
+        public void init(DirectoryPropertySource dirToWatch) throws Exception {
+            this.dirToWatch = dirToWatch;
+            File path = dirToWatch.getRootDir().toFile();
```

Comment:
you are not checking anywhere if this file exists? could this be a problem? or maybe I am missing the check?

Reply:
I have a try-catch there which would handle this case